deepchecks/utils/features.py

```diff
@@ -168,15 +168,15 @@ def calculate_feature_importance(
         importance, calc_type = _built_in_importance(model, dataset)
         # If found importance and was force permutation failure before, show warning
         if importance is not None and permutation_failure:
-            warnings.warn(permutation_failure)
+            get_logger().warning(permutation_failure)
 
     # If there was no permutation failure and no importance on the model, using permutation anyway
     if importance is None and permutation_failure is None and isinstance(dataset, tabular.Dataset):
         if isinstance(model, Pipeline):
-            pre_text = 'Cannot use model\'s built-in feature importance on a Scikit-learn Pipeline, '
+            pre_text = 'Cannot use model\'s built-in feature importance on a Scikit-learn Pipeline,'
         else:
-            pre_text = 'Could not find built-in feature importance on the model, '
-        warnings.warn(pre_text + 'using permutation feature importance calculation instead')
+            pre_text = 'Could not find built-in feature importance on the model,'
+        get_logger().warning('%s using permutation feature importance calculation instead', pre_text)
```

Comment:
what is the advantage of doing this like that and not with f-string?

Reply:
1. f-string in loggers makes pylint mad
2. logger has some smart lazy string creating so it's optimized